### ISSUE TITLE
Get rid of Tailwind CDN Warnings

### ISSUE THREAD

**Issue by kepler** [zauberzeug/nicegui#337]
Hello,

Do you have any recommended way of better using tailwind in production? Or on getting rid of the console warnings? Ideally without all the nasty npm stuff.

Their standalone CLI tool seems to be an option (https://tailwindcss.com/docs/installation), in which case I believe I'll have to turn off tailwind in `ui.run` and add my own compiled CSS.

Thanks. NiceGUI was a great finding for us and it's been a great experience in fast implementation of good looking production sites.

![image](https://user-images.githubusercontent.com/73757/216383195-07720242-16d1-4ab5-9338-534ea073fa0f.png)

/label question

**Comment by rodja** [comment]
NiceGUI delivers the full Tailwind CDN, so you have the flexibility to use all tailwind classes in a dynamic way. Therefore we could modify the `tailwind.min.js` file to remove the console warnings. But your suggestion to disable Tailwind and run the standalone CLI to generate a specialized tailwind css will hugely increase loading performance. Do not forget to include the NiceGUI package-path itself in `tailwind.config.js`. Otherwise the internally used css is not supported.
Let us know how it went. Maybe we could find a path forward where NiceGUI handles all this internally in the long run...

**Comment by rodja** [comment]
As I understand it, the main benefit of switching from `tailwind.min.js` to a compilation step is the reduced size of payload which need to be send to the client. That is of course a good reason. But I think the warning is mainly there to prevent people using the tailwindcss server for content delivery. NiceGUI uses its own copy. So the easiest way would be to remove warning from our version. Does anyone disagree? Have I missed something?

I propose to remove the warning to solve this ticket and open a discussion about how to reduce payload for tailwindcss (eg. switching to a precompile step or using something like https://github.com/mhsdesign/jit-browser-tailwindcss.

**Comment by rodja** [comment]
@kepler do you still have the issue? What would be your suggestion to move forward?

**Comment by falkoschindler** [comment]
We close this issue for now. Feel free to re-open it if you have new information.

**Comment by JS-Aibel** [comment]
> As I understand it, the main benefit of switching from `tailwind.min.js` to a compilation step is the reduced size of payload which need to be send to the client. That is of course a good reason. But I think the warning is mainly there to prevent people using the tailwindcss server for content delivery. NiceGUI uses its own copy. So the easiest way would be to remove warning from our version. Does anyone disagree? Have I missed something?
> 
> I propose to remove the warning to solve this ticket and open a discussion about how to reduce payload for tailwindcss (eg. switching to a precompile step or using something like https://github.com/mhsdesign/jit-browser-tailwindcss.

Is it possible to get an option to remove the warning?
Seeing the Console warning _“cdn.tailwindcss.com should not be used in production.”_, can for some users feel like the website is unprofessional and not secure.

**Comment by falkoschindler** [comment]
Yes, maybe we should simply remove the warning altogether. We're not using "cdn.tailwindcss.com", so it doesn't make too much sense.

It probably needs to be done in npm.py, maybe in some postprocessing step. We could basically find and remove the string `console.warn("cdn.tailwindcss.com should not be used in production. To use Tailwind CSS in production, install it as a PostCSS plugin or use the Tailwind CLI: https://tailwindcss.com/docs/installation");` in tailwindcss.min.js.

**Comment by TomFreudenberg** [comment]
I am jumping in not while getting rid of the warning but I do not want to use CDN in any way. In case of some internal rules and serving the application only internally, there should be no external request. Is there an option to define the URL for tailwindcss or do I have to work out a solution like @kepler commented in first post?

Thanks for any feedback
Tom

**Comment by rodja** [comment]
@TomFreudenberg we already serve every dependency directly -- including Tailwind. So we do not use a CDN. This PR was just about removing this misleading error message.

**Comment by TomFreudenberg** [comment]
@rodja, thanks for clarifying that!
